File: HTB-Cyber-Apocalypse-2022/web/web_acnologia_portal/web_acnologia_portal/challenge/application/bot.py
from selenium import webdriver
from flask import current_app
import time

def visit_report():
    chrome_options = webdriver.ChromeOptions()

    chrome_options.add_argument('--headless')
    chrome_options.add_argument("--incognito")
    chrome_options.add_argument('--no-sandbox')
    chrome_options.add_argument('--disable-setuid-sandbox')
    chrome_options.add_argument('--disable-gpu')
    chrome_options.add_argument('--disable-dev-shm-usage')
    chrome_options.add_argument('--disable-background-networking')
    chrome_options.add_argument('--disable-extensions')
    chrome_options.add_argument('--disable-sync')
    chrome_options.add_argument('--disable-translate')
    chrome_options.add_argument('--metrics-recording-only')
    chrome_options.add_argument('--mute-audio')
    chrome_options.add_argument('--no-first-run')
    chrome_options.add_argument('--safebrowsing-disable-auto-update')
    chrome_options.add_argument('--js-flags=--noexpose_wasm,--jitless')

    client = webdriver.Chrome(chrome_options=chrome_options)
    client.set_page_load_timeout(5)
    client.set_script_timeout(5)

    client.get('http://localhost:1337/')

    username = client.find_element_by_id('username')
    password = client.find_element_by_id('password')
    login = client.find_element_by_id('login-btn')

    username.send_keys(current_app.config['ADMIN_USERNAME'])
    password.send_keys(current_app.config['ADMIN_PASSWORD'])
    login.click()
    time.sleep(3)

    client.get('http://localhost:1337/review')

    time.sleep(3)
    client.quit()

File: 2022/HTB University 2022/web/web_batchcraft_potions/challenge/bot.js
const puppeteer = require('puppeteer');
const JWTHelper = require('./helpers/JWTHelper');
const fs        = require('fs');

const flag = fs.readFileSync('/flag.txt').toString();

const browser_options = {
    headless: true,
    executablePath: '/usr/bin/chromium-browser',
    args: [
        '--no-sandbox',
        '--disable-background-networking',
        '--disable-default-apps',
        '--disable-extensions',
        '--disable-gpu',
        '--disable-sync',
        '--disable-translate',
        '--hide-scrollbars',
        '--metrics-recording-only',
        '--mute-audio',
        '--no-first-run',
        '--safebrowsing-disable-auto-update',
        '--js-flags=--noexpose_wasm,--jitless'
    ]
};

const previewProduct = async (id) => {
    const browser = await puppeteer.launch(browser_options);

    try {
        let context = await browser.createIncognitoBrowserContext();
        let page = await context.newPage();

        let token = await JWTHelper.sign({ username: 'admin', verified: true, flag: flag });

        await page.setCookie({
            name: "session",
            value: token,
            domain: "127.0.0.1"
        });

        await Promise.race(
            [
                page.goto(`http://127.0.0.1/products/preview/${id}`, {
                    waitUntil: 'networkidle2'
                }),
                new Promise(function (reject) {
                    setTimeout(function (){
                        reject();
                    }, 7000);
                }),
            ]
        );
    }
    finally {
        await browser.close();
    }
};

module.exports = { previewProduct };